---
title: "Final Project"
author: "Tyler Rodrigues"
date: "2024-05-06"
output:
  html_document:
    toc: true
    toc_float:
      toc_collapsed: true
    toc_depth: 6
    theme: simplex
    code_folding: hide
---
\

# **Bio 476 - Final Research Project**
<p style="font-size:20pt; color:black">Lower Teakettle/Chloroflexota</p>

<!-- library(usethis) -->
<!-- use_git_config(user.name = "tgrodrigues13", user.email = "tgrodrigues@umass.edu") -->
<!-- <p style="font-size:16pt; color:black">MESSAGE</p> -->
\

```{r, eval=F, echo=F}
#'[Loading in Libraries]

if (!require("BiocManager", quietly=TRUE))
    install.packages("BiocManager")

BiocManager::install("treeio")
BiocManager::install("ggtreeExtra")
```
```{r, echo=F}
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(knitr))
suppressPackageStartupMessages(library(ggtree))
suppressPackageStartupMessages(library(TDbook))
suppressPackageStartupMessages(library(ggimage))
suppressPackageStartupMessages(library(rphylopic))
suppressPackageStartupMessages(library(treeio))
suppressPackageStartupMessages(library(tidytree))
suppressPackageStartupMessages(library(ape))
suppressPackageStartupMessages(library(TreeTools))
suppressPackageStartupMessages(library(phytools))
suppressPackageStartupMessages(library(ggnewscale))
suppressPackageStartupMessages(library(ggtreeExtra))
suppressPackageStartupMessages(library(ggstar))
suppressPackageStartupMessages(library(plotly))
```

```{r, message=F, warning=F}
#'[MAG Data]

NEON_MAGs <- read_csv("data/GOLD_Study_ID_Gs0161344_NEON_2024_4_21.csv") %>% 
#remove columns that are not needed for data analysis
  select(-c(`GOLD Study ID`, `Bin Methods`, `Created By`, `Date Added`, `Bin Lineage`)) %>% 
#create a new column with the Assembly Type
  mutate("Assembly Type" = case_when(`Genome Name` == "NEON combined assembly" ~ `Genome Name`, TRUE ~ "Individual")) %>% 
  mutate_at("Assembly Type", str_replace, "NEON combined assembly", "Combined") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "d__", "") %>%  
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "p__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "c__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "o__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "f__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "g__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "s__", "") %>%
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), ";", remove=FALSE) %>%
  mutate_at("Domain", na_if,"") %>% 
  mutate_at("Phylum", na_if,"") %>% 
  mutate_at("Class", na_if,"") %>% 
  mutate_at("Order", na_if,"") %>% 
  mutate_at("Family", na_if,"") %>% 
  mutate_at("Genus", na_if,"") %>% 
  mutate_at("Species", na_if,"") %>% 
#Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
#Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
#Get rid of the the common string "S-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
#separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
#separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-")
```

```{r, message=F, warning=F}
#'[Metagenome Data]

NEON_metagenomes <- read_tsv("data/exported_img_data_Gs0161344_NEON.tsv") %>% 
  select(-c(`Domain`, `Sequencing Status`, `Sequencing Center`)) %>% 
  rename(`Genome Name` = `Genome Name / Sample Name`) %>% 
  filter(str_detect(`Genome Name`, 're-annotation', negate = T)) %>% 
  filter(str_detect(`Genome Name`, 'WREF plot', negate = T))

#'[Reformatting Genome Name]

NEON_metagenomes <- NEON_metagenomes %>% 
#Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
#Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
#Get rid of the the common string "-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
#separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
#separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-") %>% 
#filter for Site ID
  filter(`Site ID` == "TEAK")
```
 
```{r, message=F}
#'[Chemistry Data]

NEON_chemistry <- read_tsv("data/neon_plot_soilChem1_metadata.tsv") %>% 
#remove -COMP from genomicsSampleID
  mutate_at("genomicsSampleID", str_replace, "-COMP", "") %>% 
  filter(`siteID` == "TEAK")
```

```{r}
#'[Joining MAG, Metagenome, and Chemistry into a Single Dataframe]

NEON_MAGs_metagenomes_chemistry <- NEON_MAGs %>% 
  left_join(NEON_metagenomes, by = "Sample Name") %>% 
  left_join(NEON_chemistry, by = c("Sample Name" = "genomicsSampleID")) %>% 
  rename("label" = "Bin ID")
```

\

# Abstract
<!-- Short like 5 sentences -->

\

# Motivating Reasons
<!-- Probably 1 paragraph -->

\

# Introduction
<!-- Probably 2 paragraphs -->

\

# Methods
<!-- Small description of how we got data/interpreted it -->

\

# Results
<!-- Aim for 20 graphs maybe? -->

\

### Site Graphs - Lower Teakettle, Sierra National Forest, California, USA

<!-- Circle Phylogeny for TEAK Site Showing Phylum Color Labels -->

```{r, echo=F}
#'[Reading in Tree Files from GTDB]

#tree_arc <- read.tree("data/gtdbtk.ar53.decorated.tree")
tree_bac <- read.tree("data/gtdbtk.bac120.decorated.tree")
```

```{r, results=F, echo=F}
#'[Getting Your Subtree]

#Make a vector with the internal node labels
node_vector_bac = c(tree_bac$tip.label,tree_bac$node.label)

#Search for your Phylum or Class to get the node
grep("Chloroflexota", node_vector_bac, value = TRUE)
```
```{r, results=F, echo=F}
match(grep("Chloroflexota", node_vector_bac, value = TRUE), node_vector_bac)
```

```{r, echo=F}
#'[Need to Preorder Tree Before Extracting]

tree_bac_preorder <- Preorder(tree_bac)
tree_Chloroflexota <- Subtree(tree_bac_preorder, 1712)
```

```{r, echo=F}
#'[Filter Combo Table to Taxonomic Group or Site]

NEON_combo_Chloroflexota <- NEON_MAGs_metagenomes_chemistry %>% 
  filter(Phylum == "Chloroflexota")

NEON_combo_TEAK <- NEON_MAGs_metagenomes_chemistry %>% 
  filter(`Site ID.x` == "TEAK") %>% 
  filter(Domain == "Bacteria")
```

```{r, fig.width = 10, fig.height = 10, fig.align = 'center'}
#'[Graph Chunk]

TEAK_MAGs_label <- NEON_combo_TEAK$label

tree_bac_TEAK_MAGs <-drop.tip(tree_bac,tree_bac$tip.label[-match(TEAK_MAGs_label, tree_bac$tip.label)])

ggtree(tree_bac_TEAK_MAGs, layout="circular") %<+%
  NEON_MAGs_metagenomes_chemistry +
  geom_point(mapping=aes(fill=Phylum), size = 2.5, shape = 23) +
  ggtitle("Different Phylums Located at Lower Teakettle") +
  theme(plot.title = element_text(hjust = 0.5),(size = 30))
  #face = "bold"
```
This graph shows a circular phylogeny of the species located at the Lower Teakettle Site. 

\

### Phylum Graphs - Chloroflexota

```{r}

```

\

# Discussion
<!-- Only 1 paragraph -->

\

# Conclusion
<!-- Only 1 paragraph -->